File: recursion03/Binary.java
public class Binary{
	
	public static void main (String args[]){

		int [] example = {1, 2, 3, 4, 5, 6};
		System.out.println(search(example, 5, 0, 5));
	}

	public static int search(int [] data, int target, int start, int end){
		if (start > end){
			return -1;
		}

		else {
			int middle = (start + end) / 2;
			
			int compResult = target - data[middle];
			if (compResult == 0){
				return middle;
			}
			else if (compResult < 0){
				return search(data, target, start, middle - 1);
			}
			else{
				return search(data, target, middle + 1, end);
			}
		}
	}
}

		

File: tech-interview/search/binary.py
def search(array, target, start, end):
    print(start, end)

    if start > end:
        return None

    mid = (start + end) // 2

    if mid == 0:
        if array[mid] == target:
            return 0

        else:
            return None

    if mid >= len(array):
        return None

    guess = array[mid]

    if guess == target:
        return mid

    if target > guess:
        return search(array, target, mid + 1, end)

    if target < guess:
        return search(array, target, start, mid - 1)

    return 0

test = [1, 2, 3, 4, 5]

print(test[3:])
print(test[:3])

print(search(test, 6, 0, len(test) - 1))